### PR TITLE
Update imgui-sfml to version 2.2

### PR DESCRIPTION
**Pull request by fgallegosalido**
## Description
Basically updating to the new version og ImGui-SFML:
- Bump imgui-sfml version to 2.2.
- Using last version of Dear ImGui (v1.81).
- Added the option `imgui_demo`, so the user can decide to build the demo for Dear ImGui.

## Motivation and Context
The new version of ImGui-SFML comes with a few changes that are necessary to keep up-to-date with Dear ImGui and also fixes some bugs.

## How Has This Been Tested?
OS: Ubuntu 16.04 LTS
Conan version: 1.33.1
Python version: 3.8.3
CMake version: 3.19.5
Hooks: conan-center

### CODE REVIEW

**Comment by Croydon** [comment]
Several configurations failed. Can you have a look?

**Comment by fgallegosalido** [comment]
Mmm weird. Only the debug builds are failing. I'm gonna pin @eliasdaler just in case

**Comment by eliasdaler** [comment]
Looks like adding `DEBUG_POSTFIX` broke something. It was only applied for shared libs for some reason.

Can you temporarily point Conan to this commit? https://github.com/eliasdaler/imgui-sfml/commit/f35da327e79a00a9b3514fc5ff4de52456f78961

1. Try it as-is and see if CI passes. It can potentially break more stuff (now only debug shared libs are broken), but it'll be good to investigate WHY it happens, because I don't see why adding a debug postfix leads to SFML libs not being found.
2. If it still fails, try to set `IMGUI_SFML_DEBUG_POSTFIX` to an empty string

**Comment by fgallegosalido** [comment]
Okay, I will change the sources for version 2.2 to the commit you told me, to see what happens

**Comment by fgallegosalido** [comment]
And if I change the sources under the 2.2 version (using the ones in the commit instead of the 2.2 ones), the recipe fails to be created. I'm not sure how to test your specific commit. Maybe with a patch file?

**Comment by eliasdaler** [comment]
Yeah, patch file should work

**Comment by Croydon** [comment]
> And if I change the sources under the 2.2 version (using the ones in the commit instead of the 2.2 ones), the recipe fails to be created

Have you changed the URL in conandata.yml and changed the `source()` method in conanfile.py to successfully extract from the new top-level directory within the download?

**Comment by fgallegosalido** [comment]
> Have you changed the URL in conandata.yml and changed the `source()` method in conanfile.py to successfully extract from the new top-level directory within the download?

I actually tried changing the URL to the specific commit in the conandata.yml, but with no luck. I then looked into the `source()` function, but I couldn't figure it out neither after trying changing the call to `tools.get()`.

It seems that, the way it's done, it doesn't like when the version number doesn't appear in the URL. Maybe there's something I'm not considering, or at least there should be a workaround for it.

**Comment by Croydon** [comment]
This should allow to test the commit in CI 

https://github.com/bincrafters/community/commit/a04d9caf0c1a63d13db0c9c2d23df62bac68eac5

**Comment by fgallegosalido** [comment]
It seems the CI is still broken with that specific commit @eliasdaler

**Comment by eliasdaler** [comment]
I'm not sure it's a problem of ImGui-SFML's CMake... Can you reproduce it locally? It's pretty strange that CMake is not able to find SFML's libs - maybe SFML Conan recipe is not functioning correctly?

**Comment by fgallegosalido** [comment]
I'm lately running a little bit out of time, so if you feel like closing this PR to further investigate what's going on, please do it so.

**Comment by ChristianIvicevic** [comment]
Are there any plans for looking into this? I am eager to update to newer versions of imgui and the SFML binding, especially for some bugfixes.

**Comment by fgallegosalido** [comment]
> Are there any plans for looking into this? I am eager to update to newer versions of imgui and the SFML binding, especially for some bugfixes.

I guess, but now it has to target 2.3. Also, as I said, I don't have much time to look into this, and the CI keeps failing for some reason in Debug builds for the latest version. As soon as that is fixed, this should be easy to merge.

**Comment by ericLemanissier** [comment]
retrigger CI

File: recipes/imgui-sfml/all/conandata.yml
Comment (ericLemanissier) [inline recipes/imgui-sfml/all/conandata.yml:22]:
```suggestion
```

File: recipes/imgui-sfml/config.yml
Comment (ericLemanissier) [inline recipes/imgui-sfml/config.yml:7]:
```suggestion
```

**Review by ericLemanissier — COMMENTED** [review]
Hopefully we should be good to go once this is removed !

File: recipes/imgui-sfml/config.yml
Comment (fgallegosalido) [reply recipes/imgui-sfml/config.yml:7]: 2.3 came out some time ago, so maybe we should add it too

**Comment by fgallegosalido** [comment]
2.3 came out some time ago, so maybe we should add it too

**Comment by ericLemanissier** [comment]
Don't hesitate to do so. Also, don't forget to add the patch in config.yml (the same patch as 2.2 should work, because there was no change on this part of the CMakeLists, nor in line numbers)

**Comment by ericLemanissier** [comment]
2.3 builds fail because of https://github.com/ocornut/imgui/issues/4171. It seems the only way to solve this is to set cmake option IMGUI_SFML_IMGUI_DEMO=True, or wait until the next release

**Comment by fgallegosalido** [comment]
Mmm I've added the sources for ImGui-SFML 2.3 along with the sources of Dear ImGui 1.83, and now CI is failing for GCC>=5 and Clang>=5. Maybe there's some issue with libstdc++?

**Comment by fgallegosalido** [comment]
> 2.3 builds fail because of [ocornut/imgui#4171](https://github.com/ocornut/imgui/issues/4171). It seems the only way to solve this is to set cmake option IMGUI_SFML_IMGUI_DEMO=True, or wait until the next release

At this point, it seems easier to just wait until the next release, as Dear ImGui tends to release every 2 months or so (unless it is a big release, like 1.80 was)

File: recipes/imgui-sfml/all/conandata.yml
Comment (ericLemanissier) [inline recipes/imgui-sfml/all/conandata.yml:22]:
```suggestion
```

File: recipes/imgui-sfml/all/conandata.yml
Comment (ericLemanissier) [inline recipes/imgui-sfml/all/conandata.yml:32]:
```suggestion
```

File: recipes/imgui-sfml/config.yml
Comment (ericLemanissier) [inline recipes/imgui-sfml/config.yml:7]:
```suggestion
```

**Review by ericLemanissier — APPROVED** [review]

**Comment by fgallegosalido** [comment]
I was on vacation and completely missed your delete suggestions. I think this is the best move until we can add ImGui-SFML 2.3. Thanks for the work!

**Comment by ericLemanissier** [comment]
No problem, don't worry!

**Pull request merged**
